# Plasmid mediated penicillin and tetracycline resistance among Neisseria gonorrhoeae isolates from Kenya

## Background

The only way to cure the Neisseria gonorrheae is by antibiotics because there is no vaccine that can be used to prevent the disease. The antibiotic resistance of this bacteria has now raised concerned because there is increasing spread of these antibiotic resistance strains of it, leading to the disease being incurable. There are several mechanisms which are known to be used by this bacteria, such as the bacteria developing resistance in the alteration targets of the bacteria, increasing the efflux and reducing the influx of the drugs in the cells of the bacteria and the expression of the enzymes that can breakdown the drugs. The disease crudely takes to drugs to be cured, which are Ceftriaxone and azithromycin. But these drugs do not always work because bacteria have continually evolved in order to evade antibiotics. The only antibiotics that have been able to cure most of the strains of this bacteria have been cephalosporin, but there have been also some strains in USA, Asia and Europe which have also been effectively resistant to these bacteria.In Kenya, studies on gonococcal antimicrobial resistance are few and data on plasmid mediated drug resistance is limited. The present study characterizes plasmid mediated resistance in N. gonorrhoeae isolates recovered from Kenya between 2013 and 2018.

## Introduction 
### Neisseria-gonorrheae
The bacteria Neisseria gonorrheae was discovered by Albert Ludwig Sigismund Neisser in 1879 who named it after his name, Neisseria. The 
Neisseria gonorrhea is a causative agent of the known gonorrhea and it’s known to explore through just like the other bacteria but now it’s as with time developed some unique features that are crucial to its pathogenesis so it’s important to study the pathogenesis of this bacteria because there are some patterns arising from this gonorrhea which are critical to the resistance of this bacteria to the ant microbial agents there is a risk arising from the gonorrhea bacteria because according to World Health Organization there is some surveillance of this which have been identified some strings have been identified to be resistant to the most of the available antibiotics so this means this the more the spread of this bacteria which is resistant to most of these antimicrobial agents the more we love more cases of untreatable gonorrhea diseases. 

According to what the World Health Organization the best way to treat this resistant bacterium is to eliminate the organisms and prevent further spread of the microorganism.
It’s a gram negative diplococci among polymorphonuclear leukocytes this has been identified true gram staining techniques. The most recent techniques involve anything to identify a particular strain within the nucleic acid and the signatures that’s mostly acid amplification tests to identify exactly the patterns or the signatures of the commonly and body based access to know exactly the sequences that are giving it the pathogenicity and the resistance adaptation to the most of the commonly available antimicrobial agents. 
To study the morphology and the growth of this bacteria, culture can be done using appropriate media in the lab, which will lead to development of cultures which can be used to examine the bacteria, and also undergo the other molecular technique like DNA extraction and sequencing. This will also be efficient to help in storage of the bacteria for the more studies related to other aspects. The culturing is often done using antimicrobial agent that inhibits the other bacteria and fungi (because the bacteria of interest is known to be resistant to most of the antimicrobial agents.
 
In developing world where most of the facilities are limited, there is high probability of not accessing the accurate data on the impact of this bacteria, because most of the cases of this sexually transmitted infectious bacteria may not be reported, and people might not also look for the medication. Statistically, the world health organization postulated a data that nearly 78million people in the year 2012 reported to be infected with this bacteria, age group being between 15-49, disregarding the number of the unreported cases. Africa is one of the region where this data was said to be most prevalence as the highest and strongholds for this infections. There are also more risk factors for the patients who go untreated, like infertility issues, and other diseases affecting pelvic e.g. inflammation of pelvic.
##Form of pathogenicity of Neisseria gonorrheae
This bacterium usually colonizes the genital mucosa and also other parts like anal mucosa, ocular and nasopharyngeal. Once the bacteria enter the genital mucosa, the body produces the antibodies against it, which leads to the damage of the genital mucosa as the body tries to fight it. This resultant disease is fatal in that it can cause ectopic pregnancy and when transmitted to the infants during birth it can lead to blindness. eight non-pathogenic commensal Neisseria species make up a substantial proportion of the human nasal and oropharyngeal flora. Although N. meningitidis can withstand dehydration, survive outside the human host for periods of time, and spread via respiratory droplet transmission. N. gonorrhoeae is unviable if dehydrated or exposed to non-physiological temperatures. The events that led to the evolution of two separate organisms that are highly similar in core genome and physiology, and yet cause markedly distinct diseases in different locations of the human body, are not yet understood. As both the commensal and pathogenic Neisseria spp. Understanding the mechanisms through which N. gonorrhoeae interacts with and evades the host immune system are necessary to facilitate better infection prevention, diagnostic development, surveillance, and the development of vaccines or new treatments.

### Transmission

Transmission is often the most understudied stage of infections and this is also true for N. gonorrhoeae infections. A successful pathogen must be able to efficiently transmit to new hosts, and as an obligate human colonizer, N. gonorrhoeae cannot survive outside the host. High-risk populations include individuals with multiple sexual partners, and individuals who have unprotected sex. Individuals are also often unaware that they are part of a larger sexual network. N. gonorrhoeae attaches to sperm and is easily transmitted from men to their partners through the ejaculates, as they contain a high number of bacteria.
Following transmission, N. gonorrhoeae establishes contact with the mucosal epithelium to replicate and ultimately transmit to new hosts. N. gonorrhoeae is primarily a mucosal colonizer, attaching to various epithelial surfaces. The bacteria uses pili to help them cause damage, once they attach on the surface of the host cells. These pili also help the bacteria to transfer genetic material, and also provides motility of the bacteria. The lipooligosaccharides in the cell wall of the bacteria stimulates an inflammatory response of the body, which causes most of the symptoms of the disease in humans.
Type IV pili are outer membrane structures that are crucial for mediating initial cellular adherence, natural transformation competence, twitching motility and immune evasion through antigenic and phase variation35-39. Adherence to the epithelial surface and subsequent pilus retraction bring the gonococci close to the cell.
When untreated gonorrhea can spread through the blood causing Disseminated Gonococcal Infection which is life threatening as it can cause arthritis.
surface.https://www.ncbi.nlm.nih.gov/corecgi/tileshop/tileshop.fcgi?p=PMC3&id=180498&s=84&r=1&c= 


### Scattered fibrinous lesions in the pharynx and larynx
 ![image](https://github.com/kiema2001/Neisseria-gonorrheae/assets/173713662/c1679295-c4a3-4209-8f91-29c8995bc829)

The resistance of the bacteria has been attributed to factors like
1.	Overuse of the antibiotics
2.	Mismanagement of the antibiotics
3.	Substandard quality of the antibiotics
4.	*Genetic mutations of bacterium*
5.	Anal/ oral sex


### Supergonorrhea
![image](https://github.com/kiema2001/Neisseria-gonorrheae/assets/173713662/fa0f6de7-2d6e-41a3-a80f-ed4c9bfb92ac)

 https://www.who.int/news-room/fact-sheets/detail/multi-drug-resistant-gonorrhoea 
 
According to WHO, 11 July 2023, there has been rapid increase in the cases of antimicrobial resistant (supergonorrhoeaea) which has reduced options of its treatment. This postulation has been done following the cases arising from African region and Western pacific regions, which also indicates that the majority of this cases is aged 15-49yrs. This cases of AMR N. gonorrheae started to appear as soon as antimicrobial medicines were introduced, at around 1930s, and has continued to expand at remarkably high rate, leading to many of the drugs which were introduced during those decades to be gradually unable to counter the effect of this bacteria.
According to WHO most of these cases for AMR for this bacteria were for pharyngeal gonorrhea, and the reasons are
1.	Drugs do not penetrate quite well in those tissues which can facilitate resistant
2.	The tissue supports normal flora, bacteria related to gonorrheae, and this can as well lead to AMR for this bacteria.
It’s also good to note that most of the pharyngeal gonorrhoeae are asymptomatic. This means that its possible for a person to stay with this disease for a longer period without actually realizing that its gonorrhea, or before it becomes a threat, and in this state it can also be transmitted to other people.
Resistance to so many treatment options, including penicillin, sulphonamides, tetracycline, quinolones and macrolides (including azithromycin), as well as so-called last line options like cephalosporin, make N. gonorrhoeae a multidrug resistant organism.

## Problem statement
Antimicrobial resistance poses a significant threat in the public health globally, as the spread of this AMR bacteria may lead to more challenges in the treatment of the disease cause by this bacteria.In Kenya, there has been limited genomics study that have been conducted regarding this threat of the increasing resistance of the AMR N. gonorrheae bacteria. A study which was done by [Kivata et al](https://ncbi.nlm.nih.gov/pmc/articles/PMC7517623/)
# Genome announcement for N. gonorrhoeae
the sequence data for this project was obtained from NCBI SRA database, which was collected by UNITED STATES ARMY MEDICAL RESEARCH DIRECTORATE-KENYA. the accession numbers for the project are PRJNA481622 and PRJNA590515, where we fetched the data by developing a simple script;
\\\ 

* from Bio import Entrez
import os
import subprocess
import xml.etree.ElementTree as ET
import pandas as pd

Email address to use NCBI Entrez
Entrez.email = "henrykiema47@gmail.com"

Function to fetch study (SRX) accession numbers for a given project
(PRJNA) accession number
def fetch_study_accessions(project_accession):
    search_result = Entrez.esearch(db="sra", term=project_accession,
retmax=1000)
    search_record = Entrez.read(search_result)
    return search_record['IdList']

#Function to fetch run (SRR) accession numbers and metadata for a
given study (SRX) accession number
def fetch_run_accessions_and_metadata(study_accession):
    fetch_result = Entrez.efetch(db="sra", id=study_accession,
rettype="xml", retmode="xml")
    fetch_record = fetch_result.read()
    fetch_result.close()

    run_accessions = []
    metadata_list = []

    root = ET.fromstring(fetch_record)
    for run_info in root.findall(".//RUN"):
        run_accession = run_info.get("accession")
        run_accessions.append(run_accession)

        metadata_dict = {
            "run_accession": run_accession,
            "study_accession": study_accession,
            "total_spots": run_info.get("total_spots"),
            "total_bases": run_info.get("total_bases"),
            "platform":
run_info.find(".//Platform/INSTRUMENT_MODEL").text if
run_info.find(".//Platform/INSTRUMENT_MODEL") is not None else "",
            "instrument": run_info.find(".//Instrument").text if
run_info.find(".//Instrument") is not None else "",
        }
        metadata_list.append(metadata_dict)

    return run_accessions, metadata_list

#Function to download FASTQ data for a given run accession number
using fasterq-dump
def download_fastq(run_accession, output_dir="sequences"):
    if not os.path.exists(output_dir):
        os.makedirs(output_dir)
    try:
        subprocess.run(['fasterq-dump', '--outdir', output_dir,
'--split-files', run_accession], check=True)
        print(f"Downloaded FASTQ files for: {run_accession}")
    except subprocess.CalledProcessError:
        print(f"Failed to download FASTQ files for: {run_accession}")

#Define the project accession numbers
project_accession_numbers = ["PRJNA481622", "PRJNA590515"]

#Fetch study accessions and run metadata for each project
all_metadata = []
for project_accession in project_accession_numbers:
    print(f"Fetching study accessions for project {project_accession}...")
    study_accessions = fetch_study_accessions(project_accession)
    print(f"Identified {len(study_accessions)} study accessions for
project {project_accession}.")

    for study_accession in study_accessions:
        print(f"Fetching run accessions and metadata for study
{study_accession}...")
        run_accessions, metadata_list =
fetch_run_accessions_and_metadata(study_accession)
        print(f"Identified {len(run_accessions)} run accessions for
study {study_accession}.")

        all_metadata.extend(metadata_list)
        for run_accession in run_accessions:
            print(f"Downloading FASTQ for run {run_accession}...")
            download_fastq(run_accession)

#Convert the list to a DataFrame
metadata_df = pd.DataFrame(all_metadata)

#Print the metadata DataFrame

#print(metadata_df) *
\\\

The script gave us a total of 82 reads (41 forward reads, 41 reverse reads).
The next step was to check the quality of our reads, and to achieve this we used a tool called fastqc in Linux, which gave us the report in html and zip extension, but the files were in fastqc format. From this FastQC reports, which is the report of the individual read, we then did the multiqc report, which compiled all those individual reports in to a single report. The report indicated quite a good quality of the reads, but there were some bases which were below phred score of 20. 
![image](https://github.com/kiema2001/Neisseria-gonorrheae/assets/173713662/69b2337e-ebea-46c8-8b02-f2f2c103f40a)
![image](https://github.com/kiema2001/Neisseria-gonorrheae/assets/173713662/5485d76a-2fd5-427a-ba05-b8efdca64184)


This gave us indication to use a trimming tool, trimmomatic, which is capable of trimming this reads and be left with the reads that satisfy the required quality. The adapters which were used during illumina sequencing were also present in the reads, and were also showing a significant indication and need to trim them. we developed a trimming script which helped us to trim only the reads which were below the phred quality requirement of score of 20. this is how the script was developed


\\\
#!/usr/bin/bash

#Define the input directory containing your FASTQ files
input_dir="/mnt/g/project/Neisseria_project/data/sequences"

#Define the output directory where trimmed files will be saved
output_dir="/mnt/g/project/Neisseria_project/data/trimmed"

#Define Trimmomatic parameters
trimmomatic_params="ILLUMINACLIP:adapters.fasta: 2:30:10 LEADING:3 TRAILING:3 SLIDINGWINDOW:4:15 MINLEN:5"

#Loop through each file in the input directory
for file_1 in "$input_dir"/*_1.fastq.gz; do
    # Extract the base filename without extension
    filename=$(basename "$file_1" _1.fastq.gz)
    
    #Run Trimmomatic for paired-end data
    file_2="$input_dir/${filename}_2.fastq.gz"
    trimmomatic PE -phred33 "$file_1" "$file_2" "$output_dir/${filename}_1.trimmed.fastq" "$output_dir/${filename}_1.unpaired.fastq" "$output_dir/${filename}_2.trimmed.fastq" "$output_dir/${filename}_2.unpaired.fastq" $trimmomatic_params
echo "Trimming, Please wait ..."
done
\\\

after trimming, we rechecked the quality of the reads again to check whether the trimming was effective,
